java/code/src/com/suse/manager/virtualization/test/TestVirtManager.java

```diff
@@ -90,4 +90,9 @@ public void updateLibvirtEngine(MinionServer minion) {
     public Optional<List<VmInfo>> getGuestsUpdatePlan(String minionId) {
         throw new UnsupportedOperationException();
     }
+
+    @Override
+    public Optional<Map<String, Boolean>> getFeatures(String minionId) {
+        return Optional.empty();
```

Comment:
this should throw the `UnsupportedOperationException` since in a test you would override what you want to mock while everything else should throw an exception to make you aware something you did not expect was used.

Reply:
right, fixed now. Nice catch